cordapp/src/main/kotlin/net/corda/plugins/CordappPlugin.kt

```diff
@@ -52,34 +52,42 @@ class CordappPlugin : Plugin<Project> {
         jarTask.doFirst {
             val attributes = jarTask.manifest.attributes
             var skip = false
-            // Corda 4 attributes support
-            if (cordapp.contract.name != null) {
-                attributes["Cordapp-Contract-Name"] = cordapp.contract.name ?: "${project.group}.${jarTask.baseName}"
-                attributes["Cordapp-Contract-Version"] = parseVersion(cordapp.contract.versionId.toString())
-                attributes["Cordapp-Contract-Vendor"] = cordapp.contract.vendor ?: UNKNOWN
-                attributes["Cordapp-Contract-Licence"] = cordapp.contract.licence ?: UNKNOWN
-                skip = true
+            // check whether metadata has been configured (not mandatory for non-flow, non-contract gradle build files)
+            if (cordapp.contract.isEmpty() && cordapp.workflow.isEmpty() && cordapp.info.isEmpty()) {
+                project.logger.warn("Cordapp metadata not defined for this gradle build file. See https://docs.corda.net/head/cordapp-build-systems.html#separation-of-cordapp-contracts-flows-and-services")
             }
-            if (cordapp.workflow.name != null) {
-                attributes["Cordapp-Workflow-Name"] = cordapp.workflow.name ?: "${project.group}.${jarTask.baseName}"
-                attributes["Cordapp-Workflow-Version"] = parseVersion(cordapp.workflow.versionId.toString())
-                attributes["Cordapp-Workflow-Vendor"] = cordapp.workflow.vendor ?: UNKNOWN
-                attributes["Cordapp-Workflow-Licence"] = cordapp.workflow.licence ?: UNKNOWN
-                skip = true
-            }
-            // Deprecated support (Corda 3)
-            if (skip && cordapp.info.name != null)
-                project.logger.warn("Ignoring deprecated 'info' attributes. Using 'contract' and 'workflow' attributes.")
-            if (!skip && cordapp.info.name != null) {
-                attributes["Name"] = cordapp.info.name ?: "${project.group}.${jarTask.baseName}"
-                attributes["Implementation-Version"] = cordapp.info.version ?: project.version
-                attributes["Implementation-Vendor"] = cordapp.info.vendor ?: UNKNOWN
-            }
-            val (targetPlatformVersion, minimumPlatformVersion) = checkPlatformVersionInfo()
-            attributes["Target-Platform-Version"] = targetPlatformVersion
-            attributes["Min-Platform-Version"] = minimumPlatformVersion
-            if (cordapp.sealing.enabled) {
-                attributes["Sealed"] = "true"
+            else {
+                // Corda 4 attributes support
+                if (!cordapp.contract.isEmpty()) {
+                    attributes["Cordapp-Contract-Name"] = cordapp.contract.name ?: "${project.group}.${jarTask.baseName}"
+                    attributes["Cordapp-Contract-Version"] = parseVersion(cordapp.contract.versionId.toString())
```

Comment:
```kotlin
cordapp.contract.versionId.toString()
```
This must be invoking `fun Any?.toString(): String`, which means that it will return either the string representation of `versionId` or a string value of `"null"`. Is this what we want?

Reply:
Addressed